### DISCUSSION TITLE
V8.7: log4j2 cannot collect log information

### DISCUSSION THREAD

**Discussion by sdhzlzhk** [apache/skywalking#7727] [Q&A] [answered]
I installed skywalking8.7+es7 with helm, and integrated it into my microservice using k8s sidecar. 
I can see the collected jvm and other information on the UI panel of skywalking, but there is no log information.
My microservice k8s configuration information is as follows:
oap-server k8s service:  skywalking-oap.skywalking:11800

- name: SW_GRPC_LOG_SERVER_HOST
    value: "skywalking-oap.skywalking"
- name: JAVA_TOOL_OPTIONS
  value: "-javaagent:/skywalking/agent/skywalking-agent.jar
  -Dskywalking.agent.instance_name=$(ZULL_POD_NAME)
  -Dskywalking.plugin.toolkit.log.transmit_formatted=false
  -Dskywalking.plugin.toolkit.log.grpc.reporter.server_host=$(SW_GRPC_LOG_SERVER_HOST)
  -Dskywalking.plugin.toolkit.log.grpc.reporter.server_port=11800
  -Dskywalking.plugin.toolkit.log.grpc.reporter.max_message_size=10485760
  -Dskywalking.plugin.toolkit.log.grpc.reporter.upstream_timeout=30"
- name: SW_AGENT_NAME
  value: "foam-zuul"
- name: SW_AGENT_COLLECTOR_BACKEND_SERVICES
  value: "skywalking-oap.skywalking:11800"

log4j2.xml
<Configuration status="INFO">
    <Appenders>
        <Console name="Console" target="SYSTEM_OUT">
            <PatternLayout charset="UTF-8" pattern="%d{yyyy-MM-dd HH:mm:ss:SSS} [%t] [%traceId] %-5p %c{1}:%L - %m%n"/>
        </Console>
        <Async name="Async">
            <AppenderRef ref="Console"/>
        </Async>
    </Appenders>
    <Loggers>
        <Root level="INFO">
            <AppenderRef ref="Async"/>
        </Root>
    </Loggers>
</Configuration>
The skywalking UI:
![image](https://user-images.githubusercontent.com/7436812/133588298-5a004014-b801-4f31-8055-00c2ee83c614.png)
But log is empty
![image](https://user-images.githubusercontent.com/7436812/133588417-4711ebf3-63c0-4a93-a7e6-d4cd8bb97540.png)
What's wrong!!!

**Comment by wu-sheng** [comment, marked as answer]
> What's wrong!!!

Be polite. There is no point to use exclamation points. In a community, people could help if they have time. This is not a commercial service.

___

From what you provided, do you change your log4j2 configuration, and add SkyWalking's toolkit in your dependency as the document mentioned?

**Reply by sdhzlzhk** [reply]
Thank you for your reply. I am very sorry for my improper wording. As you said, my log4j2 configuration was wrong. Later I found and fixed this problem.
